src/webgpu/api/operation/rendering/indirect_draw.spec.ts

```diff
@@ -1,13 +1,184 @@
 export const description = `
 Tests for the indirect-specific aspects of drawIndirect/drawIndexedIndirect.
-
-TODO:
-* parameter_packing - Test that the indirect draw parameters are tightly packed.
-  - offset= {0, 4, k * sizeof(args struct), k * sizeof(args struct) + 4}
-  - mode= {drawIndirect, drawIndexedIndirect}
 `;
 
+import { params, poptions } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+const filled = new Uint8Array([0, 255, 0, 255]);
+const notFilled = new Uint8Array([0, 0, 0, 0]);
+
+const kDrawIndirectParametersSize = 4 * Uint32Array.BYTES_PER_ELEMENT;
+
+g.test('basics,drawIndirect')
+  .desc(
+    `Test that the indirect draw parameters are tightly packed for drawIndirect.
+An indirectBuffer is created based on indirectOffset. The actual draw args being used indicated by the
+indirectOffset is going to draw a left bottom triangle.
+While the remaining indirectBuffer is populated with random numbers or draw args
+that draw right top triangle, both, or nothing which will fail the color check.
+The test will check render target to see if only the left bottom area is filled,
+meaning the expected draw args is uploaded correctly by the indirectBuffer and indirectOffset.
+
+Params:
+    - indirectOffset= {0, 4, k * sizeof(args struct), k * sizeof(args struct) + 4}
+    `
+  )
+  .cases(
+    params()
+      .combine(
+        poptions('indirectOffset', [
+          0,
+          Uint32Array.BYTES_PER_ELEMENT,
+          1 * kDrawIndirectParametersSize,
+          1 * kDrawIndirectParametersSize + Uint32Array.BYTES_PER_ELEMENT,
+          3 * kDrawIndirectParametersSize,
+          3 * kDrawIndirectParametersSize + Uint32Array.BYTES_PER_ELEMENT,
+          99 * kDrawIndirectParametersSize,
+          99 * kDrawIndirectParametersSize + Uint32Array.BYTES_PER_ELEMENT,
+        ] as const)
+      )
+      .expand(p => {
```

Comment:
This `.expand()` appears to only ever return one result, so it doesn't actually end up "expanding" the case list. Could this whole section (computing `indirectBuffer`) just be moved inside the test function?